### PR TITLE
[CDAP-1336] add test case for example apps

### PR DESCRIPTION
**Pull request by anew**
also adds a method to TPFS to get partition times along with their paths;
also fixes an incorrect message in wf runner.
adds test cases for examples:
- StreamConversion
- UserProfiles
- FileSetExample

### CODE REVIEW

**Comment by gokulavasan** [comment]
@anwar6953 MapReduceManager waitForFinish will return before 5 minutes if the job finishes. So 5 mins is an upper bound.

**Comment by gokulavasan** [comment]
LGTM :+1:

**Comment by anew** [comment]
Yes, Gokul, that is correct. I will keep this open to add tests also for UserProfiles and FileSet.

**Comment by anew** [comment]
Or, actually, I will start a new PR for the other tests. 
Merging.

**Comment by gokulavasan** [comment]
Yes, please merge after build completes. Thanks Andreas!

**Comment by anew** [comment]
@gokul, I pushed all three test cases to this PR. Please review (the first one has not changed since you legitimated it). Thanks

**Comment by anew** [comment]
build is running here:

File: cdap-examples/StreamConversion/src/test/java/co/cask/cdap/examples/streamconversion/StreamConversionTest.java [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+public class StreamConversionTest extends TestBase {
+
+  @Test
+  public void testStreamConversion() throws Exception {
+
+    // Deploy the PurchaseApp application
+    ApplicationManager appManager = deployApplication(StreamConversionApp.class);
+
+    // TODO: in unit tests, all schedules should be disabled at deploy time, to avoid race conditions.
+    // make sure the workflow does not get started by the schedule.
```
Comment (gokulavasan) [inline cdap-examples/StreamConversion/src/test/java/co/cask/cdap/examples/streamconversion/StreamConversionTest.java]: Is there a JIRA filed for this?

File: cdap-examples/StreamConversion/src/test/java/co/cask/cdap/examples/streamconversion/StreamConversionTest.java [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+    // send some data to the events stream
+    StreamWriter streamWriter = appManager.getStreamWriter("events");
+    streamWriter.send("15");
+    streamWriter.send("16");
+    streamWriter.send("17");
+
+    // record the current time
+    long startTime = System.currentTimeMillis();
+
+    // run the workflow
```
Comment (gokulavasan) [inline cdap-examples/StreamConversion/src/test/java/co/cask/cdap/examples/streamconversion/StreamConversionTest.java]: run the mapreduce?

File: cdap-examples/UserProfiles/src/test/java/co/cask/cdap/examples/profiles/UserProfilesTest.java [hunk 3]
```diff
@@ -0,0 +1,162 @@
...
+    Assert.assertEquals(new Long(loginTime), row.getLong("login"));
+    Assert.assertNull(row.getLong("active"));
+
+    // send an event to the stream
+    long activeTime = System.currentTimeMillis();
+    StreamWriter streamWriter = applicationManager.getStreamWriter("events");
+    streamWriter.send(new Gson().toJson(new Event(activeTime, "1234", "/some/path")));
+
+    try {
+      // Wait for the last Flowlet processing 5 events, or at most 5 seconds
```
Comment (gokulavasan) [inline cdap-examples/UserProfiles/src/test/java/co/cask/cdap/examples/profiles/UserProfilesTest.java]: processing 1 event, or at most 5 seconds?

File: cdap-examples/UserProfiles/src/test/java/co/cask/cdap/examples/profiles/UserProfilesTest.java [hunk 4]
```diff
@@ -0,0 +1,162 @@
...
+      RuntimeMetrics metrics = RuntimeStats.getFlowletMetrics("UserProfiles", "ActivityFlow", "updater");
+      metrics.waitForProcessed(1, 5, TimeUnit.SECONDS);
+    } finally {
+      flowManager.stop();
+      Assert.assertFalse(flowManager.isRunning());
+    }
+    TimeUnit.SECONDS.sleep(5);
```
Comment (gokulavasan) [inline cdap-examples/UserProfiles/src/test/java/co/cask/cdap/examples/profiles/UserProfilesTest.java]: Why do we need 5 seconds sleep?

**Comment by gokulavasan** [comment]
Small comments. Otherwise LGTM :+1:

File: cdap-examples/StreamConversion/src/test/java/co/cask/cdap/examples/streamconversion/StreamConversionTest.java (hunk 1 above)
Comment (anew) [inline cdap-examples/StreamConversion/src/test/java/co/cask/cdap/examples/streamconversion/StreamConversionTest.java]: CDAP-1365

File: cdap-examples/StreamConversion/src/test/java/co/cask/cdap/examples/streamconversion/StreamConversionTest.java (hunk 2 above)
Comment (anew) [inline cdap-examples/StreamConversion/src/test/java/co/cask/cdap/examples/streamconversion/StreamConversionTest.java]: oh yes.

File: cdap-examples/UserProfiles/src/test/java/co/cask/cdap/examples/profiles/UserProfilesTest.java (hunk 3 above)
Comment (anew) [inline cdap-examples/UserProfiles/src/test/java/co/cask/cdap/examples/profiles/UserProfilesTest.java]: ogh, copy&paste error.

File: cdap-examples/UserProfiles/src/test/java/co/cask/cdap/examples/profiles/UserProfilesTest.java (hunk 4 above)
Comment (anew) [inline cdap-examples/UserProfiles/src/test/java/co/cask/cdap/examples/profiles/UserProfilesTest.java]: we don't.

**Comment by anew** [comment]
@gokul addressed all comments and started a new build at https://builds.cask.co/browse/CDAP-RBT127-5

**Comment by gokulavasan** [comment]
@anew LGTM :+1:

**Comment by anew** [comment]
Build succeded. Merging now.

**Pull request merged**